lib/assert.js

```diff
@@ -222,13 +223,25 @@ function wrapAssertions(callbacks) {
 				pending(this, intermediate);
 				// Don't reject the returned promise, even if the assertion fails.
 				return intermediate.catch(noop);
+			};
+
+			if (promise) {
+				return handlePromise.call(this, promise);
```

Comment:
`handlePromise` is an arrow function, so `.call(this` isn't necessary. Just do `return handlePromise(promise)`.